pkg/services/baremetal/host/host.go

```diff
@@ -1202,6 +1205,142 @@ func (s *Service) actionImageInstalling(ctx context.Context) actionResult {
 	}
 }
 
+func (s *Service) actionImageInstallingCustomImageURLCommand(ctx context.Context, sshClient sshclient.Client) actionResult {
+	host := s.scope.HetznerBareMetalHost
+
+	state, logFile, err := sshClient.StateOfImageURLCommand()
+	if err != nil {
+		return actionError{err: fmt.Errorf("StateOfImageURLCommand failed: %w", err)}
+	}
+
+	duration := time.Since(host.Spec.Status.LastUpdated.Time)
+	// Please keep the number (7) in sync with the docstring of ImageURL.
+	if duration > 7*time.Minute {
+		// timeout. Something has failed.
+		msg := fmt.Sprintf("ImageURLCommand timed out after %s. Deleting machine",
+			duration.Round(time.Second).String())
+		s.scope.Logger.Error(nil, msg, "logFile", logFile)
+		conditions.MarkFalse(host, infrav1.ProvisionSucceededCondition,
+			string(host.Spec.Status.ProvisioningState), clusterv1.ConditionSeverityWarning,
+			"%s", msg)
+		return s.recordActionFailure(infrav1.FatalError, msg)
+	}
+
+	switch state {
+	case sshclient.ImageURLCommandStateRunning:
+		conditions.MarkFalse(host, infrav1.ProvisionSucceededCondition,
+			string(host.Spec.Status.ProvisioningState), clusterv1.ConditionSeverityInfo,
+			"image-url-command still running")
+		return actionContinue{delay: 10 * time.Second}
+
+	case sshclient.ImageURLCommandStateFinishedSuccessfully:
+		conditions.MarkFalse(host, infrav1.ProvisionSucceededCondition,
```

Comment:
I believe that there is no point setting the conditions here. The only longer running action is the image command and I think if the condition instead just shows the state it is the same level of information. 
The events and logs would give the necessary information around the reboot and whether the image URl command finished successfully or not

Reply:
ok, removed.